testbook/client.py

```diff
@@ -48,3 +56,38 @@ def cell_output(self, cell_index):
         """
 
         outputs = self.nb['cells'][cell_index]['outputs']
+
+    def inject(self, func, args=None):
```

Comment:
I think we need to specify what cell to inject this before or after as kwargs. We likely need to be able to reference local variables executed in earlier cells. However, I'm also fine with merging this as is and tackling local variable reference in a later PR.

Reply:
Sure, I think we can look at that in a later PR.

Also, we could include an optional argument called `prerun` which can pre-run a list of cells. Something like, 

```python
def test_foo_with_args(notebook_loader):
   with notebook_loader('/path/to/notebook') as notebook:
      notebook.inject(foo_with_args, ['value1', 'value2'], prerun=[1,3,4]).assert_output_text('You passed value1 and value2')
```